### PR TITLE
Added OSPF Opaque LSAs

### PR DESCRIPTION
**Pull request by smvoigt**
**Checklist:**

-   [x] If you are new to Scapy: I have checked [CONTRIBUTING.md](https://github.com/secdev/scapy/blob/master/CONTRIBUTING.md) (esp. section submitting-pull-requests)
-   [x] I squashed commits belonging together
-   [x] I added unit tests or explained why they are not relevant
-   [x] I executed the regression tests for Python2 and Python3 (using `tox` or, `cd test && ./run_tests_py2, cd test && ./run_tests_py3`)
-   [ ] If the PR is still not finished, please create a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)

This commit adds support for OSPF Opaque LSAs. Without this commit OSPF LSA dissection breaks when encountering type- 9,10 or 11 LSAs. 

The capture file at (https://wiki.wireshark.org/SampleCaptures?action=AttachFile&do=get&target=mpls-te.cap) has type-10 LSAs. I have used one of the packets from this file in the OSPF unit test file.

### CODE REVIEW

File: scapy/contrib/ospf.py [hunk 1]
```diff
@@ -367,6 +373,38 @@ class OSPF_NSSA_External_LSA(OSPF_External_LSA):
...
 
 
+class OSPF_Link_Scope_Opaque_LSA(OSPF_BaseLSA):
+    name = "OSPF Link Scope External LSA"
+    type = 9
+    fields_desc = [ShortField("age", 1),
+                   OSPFOptionsField(),
+                   ByteField("type", 9),
+                   IPField("id", "192.168.0.0"),
+                   IPField("adrouter", "2.2.2.2"),
```
Comment (guedou) [inline scapy/contrib/ospf.py]: Could you use addresses from https://tools.ietf.org/html/rfc5737 instead?

File: test/contrib/ospf.uts [hunk 2]
```diff
@@ -47,3 +47,19 @@ assert a.answers(b)
 pkt = Ether(dst="01:00:5e:00:00:05", src="ca:11:09:b3:00:1c")/IPv6(dst="::1", src="fe80::160c:12aa:fe7e:cd28")/OSPFv3_Hdr(src="75.1.3.1")/\
       OSPFv3_Hello(options=0x12, router="10.75.0.254", backup="10.75.0.1", neighbors=["75.1.0.1"])
 assert raw(pkt) == b'\x01\x00^\x00\x00\x05\xca\x11\t\xb3\x00\x1c\x86\xdd`\x00\x00\x00\x00(Y@\xfe\x80\x00\x00\x00\x00\x00\x00\x16\x0c\x12\xaa\xfe~\xcd(\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x03\x01\x00(K\x01\x03\x01\x00\x00\x00\x00Y\x98\x00\x00\x00\x00\x00\x00\x01\x00\x00\x12\x00\n\x00(\nK\x00\xfe\nK\x00\x01K\x01\x00\x01'
+
+= OSPFv2 Opaque lsa
```
Comment (guedou) [inline test/contrib/ospf.uts]: Could you also add basic building unit tests for the three new classes?

Such as:
```
>>> r = b'\x00\x01\x00\x0b\xc0\xa8\x00\x00\x02\x02\x02\x02\x80\x00\x00\x01d\xe9\x00\x18data'
>>> p = OSPF_AS_Scope_Opaque_LSA(r)
>>> assert p.type == 11 and p.chksum == 0x64e9
```

File: scapy/contrib/ospf.py (hunk 1 above)
Comment (smvoigt) [reply scapy/contrib/ospf.py]: These are not actually IP addresses but I have changed them anyway

File: test/contrib/ospf.uts (hunk 2 above)
Comment (smvoigt) [reply test/contrib/ospf.uts]: Done. I didn't use the checksum but I added a length test.

**Pull request merged**